pkg/apiserver/resthandler.go

```diff
@@ -269,8 +269,14 @@ func CreateResource(r rest.Creater, scope RequestScope, typer runtime.ObjectType
 			errorJSON(err, scope.Codec, w)
 			return
 		}
+		// Get name, ok if empty
+		_, name, _ := scope.Namer.Name(req)
+
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
+		if len(name) > 0 {
+			ctx = api.WithValue(ctx, "name", name)
```

Comment:
To the storage interface, a la the various things we pass in.  A method "AcceptName(obj runtime.Object, name string) error" or something.

----- Original Message -----
> >  		ctx := scope.ContextFunc(req)
> >  		ctx = api.WithNamespace(ctx, namespace)
> > +		if len(name) > 0 {
> > +			ctx = api.WithValue(ctx, "name", name)
> 
> I thought we were not supposed to add functions to API objects
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7718/files#r29612592